prov/util/src/util_ctx.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2017 Intel Corporation, Inc.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <fi_enosys.h>
+#include <fi_util.h>
+
+#define OFI_CQ_ALLOW_FLAGS_tx	(FI_SEND | FI_SELECTIVE_COMPLETION)
+#define OFI_CQ_ALLOW_FLAGS_rx	(FI_RECV | FI_SELECTIVE_COMPLETION)
+
+#define OFI_CNTR_ALLOW_FLAGS_tx	(FI_SEND | FI_READ | FI_REMOTE_READ)
+#define OFI_CNTR_ALLOW_FLAGS_rx	(FI_RECV | FI_WRITE | FI_REMOTE_WRITE)
+
+#define OFI_CQ_VALID_FLAGS_tx(flags)	(flags & FI_SEND)
+#define OFI_CQ_VALID_FLAGS_rx(flags)	(flags & FI_RECV)
+
+#define OFI_CNTR_VALID_msg_FLAGS_tx(flags)		(flags & FI_SEND)
+#define OFI_CNTR_VALID_rma_FLAGS_tx(flags)		(flags & FI_READ)
+#define OFI_CNTR_VALID_rem_rma_FLAGS_tx(flags)		(flags & FI_REMOTE_READ)
+#define OFI_CNTR_VALID_msg_FLAGS_rx(flags)		(flags & FI_RECV)
+#define OFI_CNTR_VALID_rma_FLAGS_rx(flags)		(flags & FI_WRITE)
+#define OFI_CNTR_VALID_rem_rma_FLAGS_rx(flags)		(flags & FI_REMOTE_WRITE)
```

Comment:
A lot of the above macros look unnecessary.  The macro names are longer than the checks themselves.  A direct check in the code would be preferable.